stress: failed test in cockroach/kv/kv.test: TestClientNotReady

Binary: cockroach/static-tests.tar.gz sha: https://github.com/cockroachdb/cockroach/commits/218b7f0c378f9a6d028d435f6a44c3874181b49c

Stress build found a failed test:

```
=== RUN   TestClientNotReady
I160808 04:08:24.076435 /go/src/google.golang.org/grpc/clientconn.go:454  Failed to dial 127.0.0.1:37317: connection error: desc = "transport: context canceled"; please retry.
I160808 04:08:24.077688 /go/src/google.golang.org/grpc/clientconn.go:454  Failed to dial 127.0.0.1:37317: connection error: desc = "transport: context canceled"; please retry.
E160808 04:08:24.077989 rpc/context.go:191  rpc error: code = 13 desc = connection error: desc = "transport: context canceled"
--- FAIL: TestClientNotReady (0.02s)
	send_test.go:503: unexpected error: failed to send RPC: sending to all 1 replicas failed; last error: rpc error: code = 13 desc = connection error: desc = "transport: context canceled\n```

Run Details:

```
0 runs so far, 0 failures, over 5s
0 runs so far, 0 failures, over 10s
0 runs so far, 0 failures, over 15s
0 runs so far, 0 failures, over 20s
2 runs so far, 0 failures, over 25s
16 runs so far, 0 failures, over 30s
16 runs so far, 0 failures, over 35s
16 runs so far, 0 failures, over 40s
16 runs so far, 0 failures, over 45s
17 runs so far, 0 failures, over 50s
32 runs so far, 0 failures, over 55s
32 runs so far, 0 failures, over 1m0s
32 runs so far, 0 failures, over 1m5s
32 runs so far, 0 failures, over 1m10s
32 runs so far, 0 failures, over 1m15s
48 runs so far, 0 failures, over 1m20s
48 runs so far, 0 failures, over 1m25s
48 runs so far, 0 failures, over 1m30s
48 runs so far, 0 failures, over 1m35s
48 runs so far, 0 failures, over 1m40s
64 runs so far, 0 failures, over 1m45s
64 runs so far, 0 failures, over 1m50s
64 runs so far, 0 failures, over 1m55s
64 runs so far, 0 failures, over 2m0s
64 runs so far, 0 failures, over 2m5s
80 runs so far, 0 failures, over 2m10s
80 runs so far, 0 failures, over 2m15s
80 runs so far, 0 failures, over 2m20s
80 runs so far, 0 failures, over 2m25s
80 runs so far, 0 failures, over 2m30s
96 runs so far, 0 failures, over 2m35s
96 runs so far, 0 failures, over 2m40s
96 runs so far, 0 failures, over 2m45s
96 runs so far, 0 failures, over 2m50s
96 runs so far, 0 failures, over 2m55s
112 runs so far, 0 failures, over 3m0s
112 runs so far, 0 failures, over 3m5s
112 runs so far, 0 failures, over 3m10s
112 runs so far, 0 failures, over 3m15s
112 runs so far, 0 failures, over 3m20s
128 runs so far, 0 failures, over 3m25s
128 runs so far, 0 failures, over 3m30s
128 runs so far, 0 failures, over 3m35s
128 runs so far, 0 failures, over 3m40s
128 runs so far, 0 failures, over 3m45s
143 runs so far, 0 failures, over 3m50s
144 runs so far, 0 failures, over 3m55s
144 runs so far, 0 failures, over 4m0s
144 runs so far, 0 failures, over 4m5s
144 runs so far, 0 failures, over 4m10s
158 runs so far, 0 failures, over 4m15s
160 runs so far, 0 failures, over 4m20s
160 runs so far, 0 failures, over 4m25s
160 runs so far, 0 failures, over 4m30s
160 runs so far, 0 failures, over 4m35s
171 runs so far, 0 failures, over 4m40s
176 runs so far, 0 failures, over 4m45s
176 runs so far, 0 failures, over 4m50s
176 runs so far, 0 failures, over 4m55s
176 runs so far, 0 failures, over 5m0s
180 runs so far, 0 failures, over 5m5s
192 runs so far, 0 failures, over 5m10s
192 runs so far, 0 failures, over 5m15s
192 runs so far, 0 failures, over 5m20s
192 runs so far, 0 failures, over 5m25s
193 runs so far, 0 failures, over 5m30s
207 runs so far, 0 failures, over 5m35s
208 runs so far, 0 failures, over 5m40s
208 runs so far, 0 failures, over 5m45s
208 runs so far, 0 failures, over 5m50s
209 runs so far, 0 failures, over 5m55s
216 runs so far, 0 failures, over 6m0s
224 runs so far, 0 failures, over 6m5s
224 runs so far, 0 failures, over 6m10s
224 runs so far, 0 failures, over 6m15s
224 runs so far, 0 failures, over 6m20s
225 runs so far, 0 failures, over 6m25s
240 runs so far, 0 failures, over 6m30s
240 runs so far, 0 failures, over 6m35s
240 runs so far, 0 failures, over 6m40s
240 runs so far, 0 failures, over 6m45s
241 runs so far, 0 failures, over 6m50s
252 runs so far, 0 failures, over 6m55s
256 runs so far, 0 failures, over 7m0s
256 runs so far, 0 failures, over 7m5s
256 runs so far, 0 failures, over 7m10s
257 runs so far, 0 failures, over 7m15s
259 runs completed, 1 failures, over 7m19s
FAIL
```
Please assign, take a look and update the issue accordingly.

--- a/kv/send_test.go
+++ b/kv/send_test.go
@@ -20,6 +20,7 @@ import (
 	"net"
 	"reflect"
 	"strconv"
+	"strings"
 	"testing"
 	"time"
 
@@ -477,7 +478,12 @@ func TestClientNotReady(t *testing.T) {
 		_, err := sendBatch(SendOptions{
 			Context: context.Background(),
 		}, addrs, nodeContext)
-		if !testutils.IsError(err, "connection is closing|failed fast due to transport failure") {
+		expected := strings.Join([]string{
+			"context canceled",
+			"connection is closing",
+			"failed fast due to transport failure",
+		}, "|")
+		if !testutils.IsError(err, expected) {
 			errCh <- errors.Wrap(err, "unexpected error")
 		} else {
 			close(errCh)

